src/SFA.DAS.EmployerFinance.MessageHandlers/EventHandlers/EmployerAccounts/AddedPayeSchemeEventHandler.cs

```diff
@@ -0,0 +1,23 @@
+using System.Threading.Tasks;
+using MediatR;
+using NServiceBus;
+using SFA.DAS.EmployerAccounts.Messages.Events;
+using SFA.DAS.EmployerFinance.Application.Commands.AddAccountPayeScheme;
+
+namespace SFA.DAS.EmployerFinance.MessageHandlers.EventHandlers.EmployerAccounts
+{
+    public class AddedPayeSchemeEventHandler : IHandleMessages<AddedPayeSchemeEvent>
+    {
+        private readonly IMediator _mediator;
+
+        public AddedPayeSchemeEventHandler(IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+
+        public Task Handle(AddedPayeSchemeEvent message, IMessageHandlerContext context)
+        {
+            return _mediator.Send(new AddAccountPayeSchemeCommand(message.AccountId, message.PayeRef, message.Created));
```

Comment:
We discussed this off-line, but here's the gist of our conclusions for the record...

Duplicating AccountHistory.Id as the AccountPayeScheme id in employer finance would unnecessarily couple the publishing service (MA) with employer finance (e.g. if MA decided it wanted to switch to a guid id, it would have to continue with its identity id in parallel, or it would force a change in employer finance). Although we use the source id for other entities, such as for Account, those id's are qualitatively different because they ultimately get surfaced to the UI (albeit in hashed form), so we need to have the same id across services, whereas AccountHistory id is basically an implementation detail.

We don't need chronological checks when processing add and delete messages either, because
* NServiceBus deduplicates messages (as we're using Azure SQL)
* out-of-order messages will be retried
* we don't store any properties for `AccountPayeScheme` that aren't part of its composite key (for repeated add and delete scenarios)

Only in a disaster scenario where a message eventually fails after all its retries, can the system get into an incorrect state, but we can recover by replaying messages from the error queue.

We should however manually test and ideally add automated tests to check our assumptions that this will just work :tm:.